### PR TITLE
Refactor @ckeditor/ckeditor5-engine View styles utils

### PR DESCRIPTION
**Pull request by fedemp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
- https://github.com/ckeditor/ckeditor5/tree/v32.0.0/packages/ckeditor5-engine/src/view/styles
- https://github.com/ckeditor/ckeditor5/blob/72f503139a8fdf40b41cf26960a94b725f409338/packages/ckeditor5-engine/src/view/styles/utils.js#L168
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/ckeditor__ckeditor5-engine/src/view/stylesmap.d.ts [hunk 1]
```diff
@@ -494,10 +512,10 @@ export class StylesProcessor {
 }
 
 export interface BoxSides {
```
Comment (peterblazejewicz) [inline types/ckeditor__ckeditor5-engine/src/view/stylesmap.d.ts:514]: why this specific change, allowing undefined?
https://github.com/ckeditor/ckeditor5/blob/0079571382619493de5a7ec83ffd3a17a0eabaae/packages/ckeditor5-engine/src/view/stylesmap.js#L914-L938

Comment (fedemp) [reply types/ckeditor__ckeditor5-engine/src/view/stylesmap.d.ts:514]: Here upstream return all properties set as `undefined`.

https://github.com/ckeditor/ckeditor5/blob/72f503139a8fdf40b41cf26960a94b725f409338/packages/ckeditor5-engine/src/view/styles/utils.js#L168

Comment (peterblazejewicz) [reply types/ckeditor__ckeditor5-engine/src/view/stylesmap.d.ts:514]: txh!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by fedemp** [comment]
Ready to merge

**Pull request merged**